### PR TITLE
Adding helm values to set common labels and annotations

### LINKED ISSUE #34: [Feature/Chart] Custom annotations for Operator deployment
From nifikop created by [erdrix](https://github.com/erdrix): Orange-OpenSource/nifikop#10  

## Feature Request

**Is your feature request related to a problem? Please describe.**

The aim of this feature is to allow users of the helm chart providing custom annotations for operator deployment. Via annotations, users can configure metadata processed by other parties like logs collectors, kube-janitor TTL etc...

### PR DESCRIPTION
**Pull request by r65535**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #34
| License         | Apache 2.0

### What's in this PR?

Added commonLabels and commonAnnotations to nifikop helm chart

### Additional context

Tested locally against deployment, pod, role, rolebinding, serviceaccount and service. No labels/annotations applied if both options left blank

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](docs/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)
- [x] Append changelog with changes

### CODE REVIEW

File: helm/nifikop/README.md [hunk 1]
```diff
@@ -29,6 +29,8 @@ The following tables lists the configurable parameters of the NiFi Operator Helm
 | `image.imagePullSecrets.name`    | Name of the secret to connect to docker registry                                                                                                                                     | -                        |
 | `certManager.enabled`            | Enable cert-manager integration                                                                                                                                                      | `true`                   |
 | `rbacEnable`                     | If true, create & use RBAC resources                                                                                                                                                 | `true`                   |
+| `commonLabels`                   | Labels to add to all deployed resources                                                                                                                                              | `{}`                     |
```
Comment (michael81877) [inline helm/nifikop/README.md:32]: Thoughts on just naming these `labels` and `annotations` and the description clarifies they will be added to every resource in this deployment?

Comment (r65535) [reply helm/nifikop/README.md:32]: I took inspiration from `bitnami/charts` naming conventions 😄 Happy to rename if it's overly verbose

**Comment by michael81877** [comment]
I'm not sure if this is a recent feature in Helm or not, but there are named templates you can use to inject a common named template into other templates: https://helm.sh/docs/chart_template_guide/named_templates/

This could be a nice way to define labels and annotations once in a named template and inject that named template into all of the other resources. Just a suggestion!

**Comment by r65535** [comment]
> I'm not sure if this is a recent feature in Helm or not, but there are named templates you can use to inject a common named template into other templates: https://helm.sh/docs/chart_template_guide/named_templates/
> 
> This could be a nice way to define labels and annotations once in a named template and inject that named template into all of the other resources. Just a suggestion!

Great find - had no idea this was a thing! I've migrated my PR to use named templates and retested locally 😄

File: helm/nifikop/README.md (hunk 1 above)
Comment (r65535) [reply helm/nifikop/README.md]: Renamed to `labels` and `annotations` on [58485dc](https://github.com/konpyutaika/nifikop/pull/153/commits/58485dc05ade3cf232e367d24d56862b9a6e5285)

**Comment by michael81877** [comment]
LGTM

**Review by michael81877 — APPROVED** [review]

**Review by erdrix — APPROVED** [review]

**Review by erdrix — APPROVED** [review]

**Pull request merged**